client/packages/common/src/ui/layout/tables/DataTable.tsx

```diff
@@ -122,13 +124,20 @@ export const DataTableComponent = <T extends RecordWithId>({
           }}
         >
           <HeaderRow dense={dense}>
-            {columns.map(column => (
+            {displayColumns.map(column => (
               <HeaderCell
                 dense={dense}
                 column={column}
                 key={String(column.key)}
               />
             ))}
+            {!!hiddenColumnKey && (
+              <ColumnPicker
+                columns={columns}
+                onChange={setDisplayColumns}
+                columnKey={hiddenColumnKey}
+              />
```

Comment:
nit: this takes some extra horizontal space. Could go up into the row with "Group by item" but then its not nicely integrated into the table anymore...